sources/packages/web/src/services/InstitutionService.ts

```diff
@@ -102,45 +109,71 @@ export class InstitutionService {
     return await ApiClient.InstitutionLocation.allInstitutionLocationsApi();
   }
 
-  public async users(): Promise<InstitutionUserViewModel[]> {
-    const response: InstitutionUserResDto[] = await ApiClient.Institution.getUsers();
-    const viewModels: InstitutionUserViewModel[] = response.map(
-      institutionUser => {
-        const roleArray = institutionUser.authorizations
-          .map(auth => auth.authType.role || "")
-          .filter(institutionUserRole => institutionUserRole !== "");
-        const role = roleArray.length > 0 ? roleArray.join(" ") : "-";
-        const locationArray = institutionUser.authorizations
-          .map(auth => auth.location?.name || "")
-          .filter(loc => loc !== "");
-        const userType = institutionUser.authorizations
-          .map(auth => auth.authType.type)
-          .join(" ");
-        const location = userType.toLowerCase().includes("admin")
-          ? "All"
-          : locationArray.length > 0
-          ? locationArray.join(" ")
-          : "";
-
-        const viewModel: InstitutionUserViewModel = {
-          id: institutionUser.id,
-          displayName: `${institutionUser.user.firstName} ${institutionUser.user.lastName}`,
-          email: institutionUser.user.email,
-          userName: institutionUser.user.userName,
-          userType,
-          role,
-          location,
-          isActive: institutionUser.user.isActive,
-          disableRemove:
-            AuthService.shared.userToken?.userName ===
-            institutionUser.user.userName
-              ? true
-              : false,
-        };
-        return viewModel;
-      },
+  mapUserRolesAndLocation(
+    response: InstitutionUserResDto[],
+  ): InstitutionUserViewModel[] {
+    return response.map(institutionUser => {
+      const roleArray = institutionUser.authorizations
+        .map(auth => auth.authType.role || "")
+        .filter(institutionUserRole => institutionUserRole !== "");
+      const role = roleArray.length > 0 ? roleArray.join(" ") : "-";
+      const locationArray = institutionUser.authorizations
+        .map(auth => auth.location?.name || "")
+        .filter(loc => loc !== "");
+      const userType = institutionUser.authorizations
+        .map(auth => auth.authType.type)
+        .join(" ");
+      const location = userType.toLowerCase().includes("admin")
+        ? ["All"]
+        : locationArray.length > 0
+        ? locationArray
+        : [];
+
+      const viewModel: InstitutionUserViewModel = {
+        id: institutionUser.id,
+        displayName: `${institutionUser.user.firstName} ${institutionUser.user.lastName}`,
+        email: institutionUser.user.email,
+        userName: institutionUser.user.userName,
+        userType,
+        role,
+        location,
+        isActive: institutionUser.user.isActive,
+        disableRemove:
```

Comment:
can we use like this to make it simple.
disableRemove: (AuthService.shared.userToken?.userName ===
          institutionUser.user.userName)